front/src/applications/operationalStudies/components/Scenario/AddOrEditScenarioModal.tsx

```diff
@@ -19,66 +18,83 @@ import { useNavigate } from 'react-router-dom';
 import { setSuccess } from 'reducers/main';
 import { updateScenarioID } from 'reducers/osrdconf';
 import { getInfraID, getProjectID, getStudyID } from 'reducers/osrdconf/selectors';
-import { ScenarioListResult } from 'common/api/osrdEditoastApi';
-import {
-  ELECTRICAL_PROFILE_SET_URI,
-  PROJECTS_URI,
-  SCENARIOS_URI,
-  STUDIES_URI,
-} from '../operationalStudiesConsts';
+import { ScenarioListResult, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { ScenarioRequest } from 'common/api/osrdMiddlewareApi';
 
 const scenarioTypesDefaults = {
   name: '',
   description: '',
   infra: undefined,
 };
 
-type Props = {
+type AddOrEditScenarioModalProps = {
   editionMode?: boolean;
   scenario?: ScenarioListResult;
   getScenarioTimetable?: (v: boolean) => void;
 };
 
+type SelectOptionsType = { key: number | undefined; value: string };
+
 export default function AddOrEditScenarioModal({
   editionMode = false,
   scenario,
   getScenarioTimetable,
-}: Props) {
+}: AddOrEditScenarioModalProps) {
   const { t } = useTranslation('operationalStudies/scenario');
   const { closeModal } = useContext(ModalContext);
+  const [deleteScenarioRTK] =
+    osrdEditoastApi.endpoints.deleteProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId.useMutation(
+      {}
+    );
+  const [patchScenario] =
+    osrdEditoastApi.endpoints.patchProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId.useMutation(
+      {}
+    );
+  const [postScenario] =
+    osrdEditoastApi.endpoints.postProjectsByProjectIdStudiesAndStudyIdScenarios.useMutation({});
+
   const noElectricalProfileSetOption = {
     key: undefined,
     value: t('noElectricalProfileSet'),
   };
   const [currentScenario, setCurrentScenario] = useState<ScenarioListResult>(
     scenario || scenarioTypesDefaults
   );
+  const { data: electricalProfils, isError } = osrdEditoastApi.useGetElectricalProfileSetQuery();
+
   const [displayErrors, setDisplayErrors] = useState(false);
-  const [electricalProfileSetOptions, setElectricalProfileSetOptions] = useState([
-    noElectricalProfileSetOption,
-  ]);
+  const [electricalProfileSetOptions, setElectricalProfileSetOptions] = useState<
+    SelectOptionsType[]
+  >([noElectricalProfileSetOption]);
   const dispatch = useDispatch();
   const navigate = useNavigate();
   const projectID = useSelector(getProjectID);
   const studyID = useSelector(getStudyID);
   const infraID = useSelector(getInfraID);
-  const [selectedValue, setSelectedValue] = useState(noElectricalProfileSetOption);
+  const [selectedValue, setSelectedValue] = useState<SelectOptionsType | undefined>(
+    noElectricalProfileSetOption
+  );
```

Comment:
Same here, no need for a state value
```suggestion
  const selectedValue: SelectOptionsType = useMemo(
    () =>
      electricalProfileSetOptions.find(
        (option) => option.key === currentScenario.electrical_profile_set_id
      ) || noElectricalProfileSetOption,
    [electricalProfileSetOptions, currentScenario]
  );

```

Reply:
Thank you good observation !!